modules/application/application.js

```diff
@@ -0,0 +1,170 @@
+      checkList.checked = false;
+      if (element.completed) {
+        checkList.checked = true;
+      }
+    });
+
+    this.eventDispatcher();
+  }
+
+  eventDispatcher = () => {
+    this.onclickeventDispatcher();
+    this.onsubmiteventDispatcher();
+    this.onediteventDispatcher();
+  }
+
+  onclickeventDispatcher = () => {
+    // delete task
+    const buttons = document.querySelectorAll('.delete');
+    const ref = this;
+    buttons.forEach((button, index) => {
+      button.addEventListener('click', (event) => {
+        const eventIdentifier = event.currentTarget;
+        const localArray = [];
+        let count = 1;
+        eventIdentifier.ref.todoList.forEach((element, i) => {
+          if (i !== eventIdentifier.index) {
+            eventIdentifier.ref.todoList[i].index = count;
+            localArray.push(eventIdentifier.ref.todoList[i]);
+            count += 1;
+          }
+        });
+        eventIdentifier.ref.todoList = localArray;
+        eventIdentifier.ref.onSaveList();
+        ref.updateDom();
+      });
+      button.index = index;
+      button.ref = ref;
+    }, ref);
+
+    // check task
+    const checks = document.querySelectorAll('.action_check');
+    checks.forEach((check, index) => {
+      check.addEventListener('click', (event) => {
+        const refObj = event.currentTarget;
+        if (refObj.ref.todoList[refObj.index].completed) {
+          refObj.ref.todoList[refObj.index].completed = false;
+        } else {
+          refObj.ref.todoList[refObj.index].completed = true;
+        }
+        refObj.ref.onSaveList();
+        ref.updateDom();
+      });
+      check.index = index;
+      check.ref = ref;
+    }, ref);
+
+    // clear all completed
+    const clearBtn = document.querySelector('.custom-btn');
+    clearBtn.addEventListener('click', (event) => {
+      const refObj = event.currentTarget;
+      const filteredTasks = refObj.ref.todoList.filter((item) => {
+        const state = item.completed === false;
+        return state;
+      });
+      refObj.ref.todoList = filteredTasks;
+      refObj.ref.onSaveList();
+      ref.updateDom();
+    });
+    clearBtn.ref = this;
+  }
+
+  onsubmiteventDispatcher = () => {
+    this.addTask.addEventListener('keyup', (event) => {
+      if (event.keyCode !== 13) {
+        return;
+      }
+
+      const refObj = event.currentTarget;
+      const input = refObj.ref.addTask.value;
+      if (!input.replace(/\s/g, '').length || input.length <= 0) {
+        return;
+      }
+
+      if (event.keyCode === 13) {
+        refObj.ref.todoList.push({
+          index: (refObj.ref.todoList.length + 1), description: input, completed: false,
+        });
+
+        refObj.ref.onSaveList();
+        refObj.ref.addTask.value = '';
+      }
+      refObj.ref.updateDom();
+      event.preventDefault();
+    });
+    this.addTask.ref = this;
+  }
+
+  onediteventDispatcher = () => {
```

Comment:
Some of your naming methods are a bit hard to understand and it doesn't really describe what the method wants to do, in the future make sure to give your methods simple and understandable names even someone who is not a programmer can understand like `editTask`, `deleteTasks`, `UpdateList`, ...

Reply:
Gotcha 💯 , Thanks for pointing this out I'd make changes. 😃

Reply:
When I do the review I forget and I clicked on to single comment :smile: but make changes.

Reply:
Yeah, it's fine, your review was good and quite straightforward. Thanks

Reply:
Thanks @harlexkhal 
Wish you all the best and happy coding :man_technologist: